Packs/MicrosoftGraphGroups/Integrations/MicrosoftGraphGroups/MicrosoftGraphGroups.py

```diff
@@ -63,198 +61,22 @@ def parse_outputs(groups_data: Dict[str, str]) -> Tuple[dict, dict]:
-                    msg += f' Server message: {server_msg}'
-            except Exception as err:
-                demisto.error(f'Failed parsing error response - Exception: {err}')
-            raise Exception(msg)
-        try:
-            gcloud_function_exec_id = dbot_response.headers.get('Function-Execution-Id')
-            demisto.info(f'Google Cloud Function Execution ID: {gcloud_function_exec_id}')
-            parsed_response = dbot_response.json()
-        except ValueError:
-            raise Exception(
-                'There was a problem in retrieving an updated access token.\n'
-                'The response from the Demistobot server did not contain the expected content.'
-            )
-        access_token = parsed_response.get('access_token')
-        expires_in = parsed_response.get('expires_in', 3595)
-        time_buffer = 5  # seconds by which to shorten the validity period
-        if expires_in - time_buffer > 0:
-            # err on the side of caution with a slightly shorter access token validity period
-            expires_in = expires_in - time_buffer
-
-        demisto.setIntegrationContext({
-            'access_token': access_token,
-            'valid_until': epoch_seconds() + expires_in
-        })
-        return access_token
-
-    def http_request(self, method: str, url_suffix: str = None, params: Dict = None, body: Optional[str] = None,
-                     next_link: str = None):
-        """
-        Generic request to Microsoft Graph
-        """
-        token = self.get_access_token()
-        if next_link:
-            url = next_link
-        else:
-            url = f'{self._base_url}{url_suffix}'
-
-        try:
-            response = requests.request(
-                method,
-                url,
-                headers={
-                    'Authorization': 'Bearer ' + token,
-                    'Content-Type': 'application/json',
-                    'Accept': 'application/json'
-                },
-                params=params,
-                data=body,
-                verify=self._verify,
-            )
-        except requests.exceptions.SSLError as err:
-            demisto.debug(str(err))
-            raise Exception(f'Connection error in the API call to Microsoft Graph.\n'
-                            f'Check your not secure parameter.\n\n{err}')
-        except requests.ConnectionError as err:
-            demisto.debug(str(err))
-            raise Exception(f'Connection error in the API call to Microsoft Graph.\n'
-                            f'Check your Server URL parameter.\n\n{err}')
-        try:
-            data = response.json() if response.text else {}
-            if not response.ok:
-                raise Exception(f'API call to MS Graph failed [{response.status_code}]'
-                                f' - {demisto.get(data, "error.message")}')
-            elif response.status_code == 206:  # 206 indicates Partial Content, reason will be in the warning header
```

Comment:
looks like `206` was a special case, are we handling that somehow in this implementation ?

Reply:
no. missed it. on it.